python/pyspark/sql/types.py

```diff
@@ -56,6 +56,32 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self.__eq__(other)
 
+    def from_string(self, data_type):
```

Comment:
The alias (`integer`, `raw`, `logical`) are only used for R, so we could should `_parse_datatype_json_value `